flink-runtime/src/test/java/org/apache/flink/runtime/resourcemanager/slotmanager/SlotManagerImplTest.java

```diff
@@ -753,33 +742,18 @@ public void testTaskManagerSlotRequestTimeoutHandling() throws Exception {
 
 			slotManager.registerSlotRequest(slotRequest);
 
-			ArgumentCaptor<SlotID> slotIdCaptor = ArgumentCaptor.forClass(SlotID.class);
-
-			verify(taskExecutorGateway, times(1)).requestSlot(
-				slotIdCaptor.capture(),
-				eq(jobId),
-				eq(allocationId),
-				anyString(),
-				eq(resourceManagerId),
-				any(Time.class));
-
-			TaskManagerSlot failedSlot = slotManager.getSlot(slotIdCaptor.getValue());
+			final SlotID firstSlotId = slotIds.take();
+			TaskManagerSlot failedSlot = slotManager.getSlot(firstSlotId);
 
 			// let the first attempt fail --> this should trigger a second attempt
 			slotRequestFuture1.completeExceptionally(new SlotAllocationException("Test exception."));
 
-			verify(taskExecutorGateway, times(2)).requestSlot(
-				slotIdCaptor.capture(),
-				eq(jobId),
-				eq(allocationId),
-				anyString(),
-				eq(resourceManagerId),
-				any(Time.class));
-
 			// the second attempt succeeds
 			slotRequestFuture2.complete(Acknowledge.get());
 
-			TaskManagerSlot slot = slotManager.getSlot(slotIdCaptor.getValue());
+			final SlotID secondSlotId = slotIds.take();
```

Comment:
same as above; maybe assert that it is empty now?

Reply:
Will add the empty check